src/transformations.jl

```diff
@@ -97,100 +121,135 @@ julia> StatsBase.transform(dt, X)
  -1.0   0.0  1.0
 ```
 """
-function fit(::Type{ZScoreTransform}, X::AbstractArray{<:Real,2}; center::Bool=true, scale::Bool=true)
-    d, n = size(X)
-    n >= 2 || error("X must contain at least two columns.")
-
+function fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real};
+             dims::Union{Integer,Nothing}=nothing, center::Bool=true, scale::Bool=true)
+    if dims == 1
+        n, l = size(X)
+        n >= 2 || error("X must contain at least two rows.")
+        m, s = mean_and_std(X, 1)
+    elseif dims == 2
+        l, n = size(X)
+        n >= 2 || error("X must contain at least two columns.")
+        m, s = mean_and_std(X, 2)
+    elseif dims === nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
+        m, s = mean_and_std(X, 2)
+    else
+        throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
+    end
     T = eltype(X)
-    m, s = mean_and_std(X, 2)
-
-    return ZScoreTransform(d, (center ? vec(m) : zeros(T, 0)),
-                              (scale ? vec(s) : zeros(T, 0)))
+    return ZScoreTransform(l, dims, (center ? vec(m) : zeros(T, 0)),
+                                    (scale ? vec(s) : zeros(T, 0)))
 end
 
-function transform!(y::AbstractVecOrMat{<:Real}, t::ZScoreTransform, x::AbstractVecOrMat{<:Real})
-    d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(y,2)
-    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+function fit(::Type{ZScoreTransform}, X::AbstractVector{<:Real};
+             dims::Union{Integer,Nothing}=nothing, center::Bool=true, scale::Bool=true)
+    if dims == nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
+    elseif dims != 1
+        throw(DomainError(dims, "fit only accept dims=1 over a vector. Try fit(t, x, dims=1)."))
+    end
 
-    m = t.mean
-    s = t.scale
+    T = eltype(X)
+    m, s = mean_and_std(X)
+    return ZScoreTransform(1, dims, (center ? [m] : zeros(T, 0)),
```

Comment:
Has this been addressed?

Reply:
Ah, OK, it's just meant to create an empty vector. It would be more logical written as `T[]`, but that doesn't matter a lot.

Do you think the PR is ready for another (hopefully the last) round of review?

Reply:
Sure, anytime.

Reply:
Sure, anytime.